src/umpire/Umpire.cpp

```diff
@@ -255,10 +255,17 @@ void* find_pointer_from_name(Allocator allocator, const std::string& name)
 }
 
 #if defined(UMPIRE_ENABLE_MPI)
-MPI_Comm get_communicator_for_allocator(Allocator a, MPI_Comm comm)
+namespace {
+static std::map<int, MPI_Comm> cached_communicators{};
```

Comment:
I would move this inside the get_cached_communicators() function.

Reply:
Ah yes, good idea